### PR TITLE
IDE-534 Bookmarks not working on local files

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixes IDE-534

Signed-off-by: dehilsterlexis <david.dehilster@lexisnexis.com>

### CODE REVIEW

File: eclide/BookmarksFrame.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ class CBookmarksFrame :
     void Post_Refresh();
     bool OnDoubleClick();
     bool OnContext(CMenu &/*menu*/, CPoint &/*pt*/);
-    void ParseBookmarksEcl(std::_tstring ecl, std::_tstring user, std::_tstring inModule, std::_tstring inAttributeName, IAttributeType *attrType);
+    void ParseBookmarksEcl(std::_tstring ecl, std::_tstring user, std::_tstring id, std::_tstring inModule, std::_tstring inAttributeName, IAttributeType *attrType);
```
Comment (GordonSmith) [inline eclide/BookmarksFrame.h]: Minor:  These should be ```const std::_tstring & ecl, ...``` declarations

File: eclide/BookmarksView.cpp [hunk 2]
```diff
@@ -504,6 +558,10 @@ boost::filesystem::path CBookmarksView::BookmarksFilePath()
 {
     boost::filesystem::path fullpath;
     GetUserFolder(fullpath, static_cast<const TCHAR *>(CString(GetIConfig(QUERYBUILDER_CFG)->Get(GLOBAL_USER))));
+    if (IsLocalRepositoryEnabled() == TRI_BOOL_TRUE)
+    {
+        return fullpath / BOOKMARKS_FILE_LOCAL;
+    }
```
Comment (GordonSmith) [inline eclide/BookmarksView.cpp]: this ```if``` is not needed

File: eclide/ChildBuilderFrame.cpp [hunk 3]
```diff
@@ -1047,7 +1047,7 @@ END_MESSAGE_MAP()
 
 // CChildBuilderFrm message handlers
 //  ===========================================================================
-bool RestoreExisting(IWorkspaceItem * workspaceItem, CChildBuilderFrm* pChild)
+bool RestoreExisting(IWorkspaceItem * workspaceItem, CChildBuilderFrm* &pChild)
```
Comment (GordonSmith) [inline eclide/ChildBuilderFrame.cpp]: Change it to:
```
CChildBuilderFrm** pChild
```

And change the callers to:
```    
if (!RestoreExisting(workspaceItem, &pChild))
```

File: eclide/ChildBuilderFrame.cpp [hunk 4]
```diff
@@ -1124,6 +1124,26 @@ HWND OpenBuilderMDI(CMainFrame* pFrame, IAttribute *src, IWorkspaceItem * worksp
...
+        CreateNewChild(pFrame, pChild, IDR_BUILDERWINDOW, workspaceItem->GetLabel());
+        if (!pChild->m_view->DoFileOpen(filePath))
+        {
+            pChild->DestroyWindow();
+            return false;
+        }
+        pChild->m_view->SetReadOnly(locked);
+    }
+    unsigned int pos = pChild->m_view->m_dlgview.m_view.PositionFromLine(row - 1) + col - 1;
+    pChild->m_view->m_dlgview.m_view.SetSel(pos, pos + len);
```
Comment (GordonSmith) [inline eclide/ChildBuilderFrame.cpp]: Add a method to SourceView.h ```SetSel``` and then internally pass the call onto m_dlgView (IOW pretend m_dlgView is private/protected.

File: eclide/EclDlgBuilder.cpp [hunk 5]
```diff
@@ -89,6 +101,16 @@ bool CBuilderDlg::DoSave(bool attrOnly)
...
 
+CBookmarksFrame *CBuilderDlg::GetBookmarksFrame() {
+    CBookmarksFrame *pFrame = (CBookmarksFrame *)GetIMainFrame()->m_Bookmarks;
+    if (!pFrame)
+    {
+        return NULL;
+    }
+
+    return pFrame;
+}
```
Comment (GordonSmith) [inline eclide/EclDlgBuilder.cpp]: Could imply be:  ```return (CBookmarksFrame *)GetIMainFrame()->m_Bookmarks;```?

**Comment by dehilsterlexis** [comment]
Closing because I made my repository public from being private and now the pushes are not coming here. Will open IDE-534P (P for public).

**Pull request closed without merging**